ui/packages/shared/profile/src/components/ProfileShareButton/index.tsx

```diff
@@ -0,0 +1,114 @@
+import {useState} from 'react';
+import {Button, Modal} from '@parca/components';
+import {Icon} from '@iconify/react';
+import {QueryRequest, QueryServiceClient} from '@parca/client';
+import ResultBox from './ResultBox';
+
+interface Props {
+  queryRequest: QueryRequest;
+  queryClient: QueryServiceClient;
+}
+
+interface ProfileShareModalProps {
+  queryRequest: QueryRequest;
+  queryClient: QueryServiceClient;
+  isOpen: boolean;
+  closeModal: () => void;
+}
+
+const ProfileShareModal = ({
+  isOpen,
+  closeModal,
+  queryRequest,
+  queryClient,
+}: ProfileShareModalProps) => {
+  const [isShared, setIsShared] = useState(false);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string>('');
+  const [description, setDescription] = useState<string>('');
+  const [sharedLink, setSharedLink] = useState<string>('');
+  const isFormDataValid = () => true;
+
+  const handleSubmit: () => void = async () => {
+    try {
+      setLoading(true);
+      const {response} = await queryClient.shareProfile({queryRequest, description});
+      setSharedLink(response.link);
+      setLoading(false);
+      setIsShared(true);
+    } catch (err) {
+      console.error(err);
+      setLoading(false);
+      setError(err.toString());
+    }
+  };
+
+  const onClose = () => {
+    setLoading(false);
+    setError('');
+    setIsShared(false);
+    closeModal();
+  };
+
+  return (
+    <Modal isOpen={isOpen} closeModal={onClose} title="Share Profile" className="w-[420px]">
+      <form className="py-2">
+        <p className="text-sm text-gray-500">
+          Note: Shared profiles can be accessed by any one with the link, even from people outside
+          your organisation.
+        </p>
+        {!isShared || error?.length > 0 ? (
+          <>
+            <p className="text-sm text-gray-500 mt-3 mb-2">Enter a description(optional)</p>
+            <textarea
+              className="border w-full"
```

Comment:
```suggestion
              className="border w-full text-gray-500 text-sm px-2 py-2"
```

I added these style changes cos I think it makes the `textarea` field look better.

*Before* 
<img width="474" alt="image" src="https://user-images.githubusercontent.com/9016992/177750911-421d1395-dd41-4f2c-8b8d-228f6d5e0545.png">


*After*
<img width="478" alt="image" src="https://user-images.githubusercontent.com/9016992/177749299-e10be05e-185b-43eb-9b86-0cd1cbd3a8aa.png">

Reply:
Thanks for the suggestions, Yomi. Applied them. 

Also pushed a commit to add dark mode support for the Modal in this commit: https://github.com/parca-dev/parca/pull/1313/commits/cce0ed6edb66c7edeb18300ed1f0e7376ec17a08

I had moved the `dark` and other theme classes to `html` and `body` tags to support the `Modal` component, as it is being created as a React portal outside the `id="root"` element. 
Please let me know if you have any concerns.

Reply:
Yep, I'm planning to do the same fix for the cloud app as well. 👍🏼